crates/iceberg/src/arrow/value.rs

```diff
@@ -440,10 +440,12 @@ impl PartnerAccessor<ArrayRef> for ArrowArrayAccessor {
         Ok(schema_partner)
     }
 
+    // todo generate field_pos in datafusion instead of passing to here
```

Comment:
I found it tricky to handle this case: the input from datafusion won't have field id, and we will need to assign them manually. maybe there is a way to do name mapping here?

Reply:
This is a temporary hack to an issue that I don't know how exactly to fix for now: the `RecordBatch` from Datafusion won't have `PARQUET_FIELD_ID_META_KEY` in its schema's metadata, causing the schema visiting to fail [here](https://github.com/apache/iceberg-rust/blob/fbc3716c7eac6bba6f1902610407e82e925a83ba/crates/iceberg/src/arrow/schema.rs#L244).

I'm thinking maybe we can bound the schema in datafusion via name mapping, but have not got the chance to explore more

Reply:
This method is used when using `ParquetWriter` to write `RecordBatch`. When it's counting nan values, it will need to walk through both `RecordBatch`'s schema and Iceberg schema in a partner fashion: https://github.com/apache/iceberg-rust/blob/9787140165a15afaf50fc4742484d22c2230bd60/crates/iceberg/src/writer/file_writer/parquet_writer.rs#L528

Basically the call stack is `NanValueCountVisitor::compute` -> `visit_struct_with_partner` -> `ArrowArrayAccessor::field_partner` -> `get_field_id`

Reply:
Created an issue for this: https://github.com/apache/iceberg-rust/issues/1560